src/libutils/oglapphelpers/oglapp.cpp

```diff
@@ -111,6 +111,11 @@ void OglApp::redisplay()
         std::swap(pts[1], pts[3]);
     }
 
+    // Bind OpenGL resources before drawing
+
+    m_oglBuilder->useAllTextures();
```

Comment:
Should the calls at the end of setShader be removed?

Reply:
Maybe updateUniforms() should be removed.

Reply:
> Should the calls at the end of setShader be removed?

My guess is yes, assuming this behavior is not used somewhere else, but I'd say it would be an error.

> Maybe updateUniforms() should be removed.

Do you mean removing `::updateUniforms()` method but leaving the `m_oglBuilder->useAllUniforms();` call in `::redisplay()` ?

My first impression would be that ociochecklut needs test first, as in without this PR it won't produce any results with the --gpu or --gpulegacy flags I think. Then even with --gpu working, you would still get false results without the proper texture bindings. Looking at the code, there is no tests for OCIO apps currently no ?

Reply:
> Looking at the code, there is no tests for OCIO apps currently no ?

That's true. But we are open to external contributors with some time to build unit tests for the command line tools :-)

Reply:
The update of the uniforms is definitively missing from `OglApp::redisplay()`. But it should use `OglApp::updateUniforms()` to do it.

Reply:
Doesn't the texture content changes for each test values you want to check the LUT for (ig. `--t`) ?

`m_oglBuilder->useAllTextures();` seems to be doing both binding and uniform setup, agree that the naming is not obvious.

I'll update to `OglApp::updateUniforms()` then, thanks.

Reply:
My point is that if uniforms are always updated when a rediaply is done, why would another call to update the uniforms be needed.

Reply:
> My point is that if uniforms are always updated when a redisplay() is done, why would another call to update the uniforms be needed?

That's a good point. The public method (i.e. `OglApp::updateUniforms()`) is then useless.

> Doesn't the texture content changes for each test values you want to check the LUT for (ig. --t) ?

Only the input texture (i.e. the image to process) changes. The textures related to the color transformation such as those created for the LUTs, do not. And `m_oglBuilder->useAllTextures();` only binds & updates the latter.